snooty/parser.py

```diff
@@ -608,6 +609,12 @@ def handle_directive(
 
             self.validate_and_add_asset(doc, argument_text, line)
 
+        elif name == "cta-banner":
+            if options is None:
+                self.diagnostics.append(
+                    MissingRequiredOption("cta-banner", "url", line)
```

Comment:
We may not need this diagnostic / options check. Omitting the url option results in a `DocUtilsParseError` and prints out:

```
ERROR(documents.txt:32ish): "cta-banner" requires the following options: url
```

Reply:
Oh, good catch! I forgot that we supported required options via the `required = true` option in rstspec